interview-questions.md

```diff
@@ -6,50 +6,55 @@ First, without external resources. Challenge yourself to answer from memory as i
 
-Researched answer:
+Researched answer: Components let a developer split the application into separate and reusable pieces. Components can be thought of as similar to functions in javascript, since functions can be re-used as many times as the user wishes. Components accept inputs from the user, which are called properties, from the main component App.js. These inputs are passed into the component as properties, or props.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are code that can be changed over time, depending on the input of the user. Once can then create a handleChange function, and make that function change the state of the state variable. Properties, or props, are then passing that change into another component, or the child. We can do so by including a component call in our App component, and including the state variable inside of the component call. We then go into said component and call it using props.stateVariable.
 
-Researched answer:
+Researched answer: props are used to trasnfer data from parent component to child component. State is owned locally, and the component itself updates it. Props are owned and read by the parent. A prop can then be passed down to the child from the parent's state function. All the work, however, is done by the parent, and the child only displays that. 
 
 5. What is the DOM?
 
-Your answer:
+Your answer: dom is Document Object Model. It is a piece of code that is always listening for changes that the user provides. That is all that I remember from the lecture. 
 
-Researched answer:
+Researched answer: DOM allows javascript to access snippets of HTML more easily. The DOM represents the entire application's user interface; DOM is usually represented by a tree structure. 
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: A div tag is a tag in HTML that is normally used to visually divide sections of HTML in HTML code. div tags are useful for giving class names. These class names can then be used to make visual changes and make applications more visually pleasing. A span tag is a tag which spans the whole page.
 
-Researched answer:
+Researched answer: div tags allow modifications to be made for the whole block that is enclosed in the div tag. A span tag is used more for making modifications in a line, or a sentence. 
```

Comment:
Nice follow-up with the span tag! It allows us to target a word or words as opposed to the whole block like a div.